### PR TITLE
Stop reporting error metrics from KMC if secrets or shoots are not found

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Changes proposed in this pull request:

- Stop reporting error metrics from KMC if secrets or shoots are not found.
- Fix the development mode deployment.

**Related issue(s)**

- https://github.com/kyma-project/control-plane/issues/1640

### CODE REVIEW

**Review by klaudiagrz — APPROVED** [review]

File: components/kyma-metrics-collector/pkg/gardener/shoot/client.go [hunk 1]
```diff
@@ -32,12 +33,17 @@ func NewClient(opts *options.Options) (*Client, error) {
...
+
+	if err == nil {
+		gardenercommons.TotalCalls.WithLabelValues(gardenercommons.SuccessStatusLabel, shootName, gardenercommons.SuccessGettingShootLabel).Inc()
+		return convertRuntimeObjToShoot(unstructuredShoot)
+	}
+
+	if !errors.IsNotFound(err) {
 		gardenercommons.TotalCalls.WithLabelValues(gardenercommons.FailureStatusLabel, shootName, gardenercommons.FailedGettingShootLabel).Inc()
-		return nil, err
 	}
```
Comment (mfaizanse) [inline components/kyma-metrics-collector/pkg/gardener/shoot/client.go:44]: Should we log an error message if it's a NotFound error?

File: components/kyma-metrics-collector/pkg/gardener/shoot/client_test.go [hunk 2]
```diff
@@ -43,10 +43,10 @@ func TestGet(t *testing.T) {
 	g.Expect(err).ShouldNot(gomega.BeNil())
 	g.Expect(k8sErrors.IsNotFound(err)).To(gomega.BeTrue())
 	// Test metric
-	g.Expect(testutil.CollectAndCount(gardenercommons.TotalCalls, metricName)).Should(gomega.Equal(2))
+	g.Expect(testutil.CollectAndCount(gardenercommons.TotalCalls, metricName)).Should(gomega.Equal(1))
 	callsFailure, err := gardenercommons.TotalCalls.GetMetricWithLabelValues(gardenercommons.FailureStatusLabel, nonexistentShoot, gardenercommons.FailedGettingShootLabel)
 	g.Expect(err).Should(gomega.BeNil())
-	g.Expect(testutil.ToFloat64(callsFailure)).Should(gomega.Equal(float64(1)))
+	g.Expect(testutil.ToFloat64(callsFailure)).Should(gomega.Equal(float64(0)))
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/kyma-metrics-collector/pkg/gardener/shoot/client_test.go:49]: In the client logic we ignore reporting metrics for 404 errors, which is easy to simulate in the tests.
Not sure how to simulate non-404 errors in the tests. Do you have any suggestions?

File: components/kyma-metrics-collector/pkg/gardener/shoot/client.go (hunk 1 above)
Comment (marcobebway) [reply components/kyma-metrics-collector/pkg/gardener/shoot/client.go:44]: It's already logged one level above (see: https://github.com/kyma-project/control-plane/blob/2093b3b930baada408b5622ca8d801f5201a3e06/components/kyma-metrics-collector/pkg/process/process.go#L98)

File: components/kyma-metrics-collector/pkg/gardener/shoot/client_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mfaizanse) [reply components/kyma-metrics-collector/pkg/gardener/shoot/client_test.go:49]: not sure, is there any auth in the client call? may an invalid auth token or something!!!

[the remark this answers is not in the document]
Comment (marcobebway) [reply components/kyma-metrics-collector/pkg/gardener/shoot/client_test.go:49]: This also does not seem possible with the fake client.

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**